src/classes/BGE_ConfigurationWizard_CTRL.cls

```diff
@@ -36,6 +36,46 @@
 */
 public with sharing class BGE_ConfigurationWizard_CTRL {
 
+    /*******************************************************************************************************
+    * @description These fields are required only if at least 1 field from its object group is selected
+    */
+    private static Set<String> systemRequiredFields = new Set<String>{
+            'Account1_Name__c',
+            'Account2_Name__c',
+            'Donation_Amount__c',
+            'Contact1_LastName__c',
+            'Contact2_LastName__c'};
+
+    /*******************************************************************************************************
+    * @description Checks for field permissions to create a new Batch/Template
+    * @param sObjectName string of the sObjectName retrieved with force:hasSObjectName
+    * @return void; throws an AuraHandledException if any fields fail FLS check
+    */
+    @AuraEnabled
+    public static void checkFieldPermissions(String sObjectName) {
+
+        List<DescribeFieldResult> fieldsToCheck = new List<DescribeFieldResult>();
+
+        List<String> fields = getRecordFieldNames(sObjectName);
+        for (String field : fields) {
+            if (field != 'Id') {
+                fieldsToCheck.add(UTIL_Describe.getFieldDescribe(sObjectName, UTIL_Namespace.StrTokenNSPrefix(field)));
+            }
+        }
+
+        // Donation Amount is also required
+        fieldsToCheck.add(UTIL_Describe.getFieldDescribe(UTIL_Namespace.StrTokenNSPrefix('DataImport__c'), UTIL_Namespace.StrTokenNSPrefix('Donation_Amount__c')));
```

Comment:
They just wouldn't see any. That's a valid use case if the org doesn't use Payments.

Reply:
Instead of making this public, let's move this to two methods:

```
checkBatchFieldPermissions()
checkTemplateFieldPermissions() 
```

That makes the method calls clearer.